add calibration analysis doc (#554)
neulab/ExplainaBoard: docs/add_custom_features.md

where

* `feature-level`: the fine-grained level of the analysis
  * `example` for sentence-level analysis
  * `span` for span-level analysis (e.g. in named entity recognition analysis)
  * `token` for token-level analysis (e.g. in named entity conditional text generation)
* `features`: a list of feature names where each feature are predefined. These
  features can be true_label, predicted_label, or other custom features.

## Example of CalibrationAnalysis

> CalibrationAnalysis is to measure the descrepency between the predicting
> probability and the true correctness likelihood. It must come together
> with the accuracy metric. For tasks with a default accuracy metric,
> calibration analysis is automatically perform on `confidence` feature.
> You can directly add the `confidence` values to the examples in the json output.
> You may also customize calibration analysis with custom feature name and settings.

Reviewer: ```suggestion
CalibrationAnalysis measures [calibration](https://arxiv.org/abs/1706.04599), the association 
between a model-predicted probability and the likelihood of the answer being correct.
It must be used on tasks that use the Accuracy metric. For tasks that use the Accuracy metric,
by default, calibration analysis is automatically performed if your model inputs have the `confidence` feature.
You can directly add the `confidence` values to the examples in a json-formatted system output file.
You may also customize calibration analysis with custom feature name and settings.
```

> Default calibration analysis format [example](../data/system_outputs/absa/absa-example-output-confidence.json).

Reviewer: ```suggestion
Default calibration analysis format [example](../data/system_outputs/absa/absa-example-output-confidence.json):
```

```json
{
  "examples": [
    {
      "predicted_label": "positive",
      "confidence": 0.22101897026820283
    }
  ]
}
```

> Custom calibration analysis format [example](../data/system_outputs/absa/absa-example-output-custom-calibration-analysis.json).

Reviewer: ```suggestion
Custom calibration analysis format [example](../data/system_outputs/absa/absa-example-output-custom-calibration-analysis.json):
```

```json
{
  "metadata": {
    "custom_features": {
      "example": {

>         "log_likelihood": {

Reviewer: This isn't the log likelihood because it's positive. Maybe you just mean `probability`?
